### PR TITLE
Add arg `maintain_order` to `unique()`

### PR DESCRIPTION
**Pull request by etiennebacher**
Follow-up of #237

### CODE REVIEW

File: tests/testthat/test-lazy.R [hunk 1]
```diff
@@ -247,6 +247,21 @@ test_that("unique", {
...
+    y = rep(c("A", "B", "C"), 2)
+  )
+  # with maintain_order = FALSE, the order could still be maintained once in
+  # a while due to chance so we repeat this test several times
+  for (i in 1:5) {
+    expect_equal(
+      df$lazy()$unique(maintain_order = TRUE)$collect()$to_data_frame()$x,
+      c("A", "B", "C")
+    )
+  }
```
Comment (eitsupi) [inline tests/testthat/test-lazy.R:262]: I don't see the validity of repeating 5 times here.
Is it possible to reduce the chance factor by fixing seed or setting some environment variable?

File: tests/testthat/test-dataframe.R [hunk 2]
```diff
@@ -598,6 +598,21 @@ test_that("unique", {
...
+    y = rep(c("A", "B", "C"), 2)
+  )
+  # with maintain_order = FALSE, the order could still be maintained once in
+  # a while due to chance so we repeat this test several times
+  for (i in 1:5) {
+    expect_equal(
+      df$unique(maintain_order = TRUE)$to_data_frame()$x,
+      c("A", "B", "C")
+    )
+  }
```
Comment (eitsupi) [inline tests/testthat/test-dataframe.R:613]: ditto

**Review by eitsupi — COMMENTED** [review]
Looks good, thanks.

One comment about the tests.

File: tests/testthat/test-lazy.R (hunk 1 above)
Comment (etiennebacher) [reply tests/testthat/test-lazy.R:262]: I don't know how to do that, @sorhawell do you have some idea?

Comment (sorhawell) [reply tests/testthat/test-lazy.R:262]: We could test on  a column of e.g. a hundred unique elements, then the risk if randomly getting the order right by chance would be astronomically low `1 / 100! = 1E-157`.

**Comment by sorhawell** [comment]
I will add some minor changes to rust code tonight

File: tests/testthat/test-lazy.R (hunk 1 above)
Comment (etiennebacher) [reply tests/testthat/test-lazy.R]: indeed, that was a very easy solution :smile:

**Review by sorhawell — APPROVED** [review]

**Pull request merged**